pkg/network/tracer/cilium_lb.go

```diff
@@ -0,0 +1,270 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build linux_bpf
+
+package tracer
+
+import (
+	"context"
+	"encoding/binary"
+	"errors"
+	"fmt"
+	"net"
+	"sync"
+	"time"
+	"unsafe"
+
+	"github.com/cilium/ebpf"
+	"github.com/prometheus/client_golang/prometheus"
+	"golang.org/x/sys/unix"
+
+	"github.com/DataDog/datadog-agent/pkg/ebpf/maps"
+	"github.com/DataDog/datadog-agent/pkg/network"
+	"github.com/DataDog/datadog-agent/pkg/network/config"
+	"github.com/DataDog/datadog-agent/pkg/network/netlink"
+	"github.com/DataDog/datadog-agent/pkg/process/util"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+const (
+	//revive:disable
+	// from github.com/cilium/cilium/pkg/maps/ctmap/lookup.go
+	TUPLE_F_OUT     = 0
+	TUPLE_F_IN      = 1
+	TUPLE_F_RELATED = 2
+	TUPLE_F_SERVICE = 4
+	//revive:enable
+)
+
+type tupleKey4 struct {
+	DestAddr   [4]byte
+	SourceAddr [4]byte
+	SourcePort uint16
+	DestPort   uint16
+	NextHeader uint8
+	Flags      uint8
+}
+
+type ctEntry struct {
+	Reserved0 uint64
+	BackendID uint64
+	Packets   uint64
+	Bytes     uint64
+	Lifetime  uint32
+	Flags     uint16
+	// RevNAT is in network byte order
+	RevNAT           uint16
+	IfIndex          uint16
+	TxFlagsSeen      uint8
+	RxFlagsSeen      uint8
+	SourceSecurityID uint32
+	LastTxReport     uint32
+	LastRxReport     uint32
+}
+
+type backend4KeyV3 struct {
+	ID uint32
+}
+
+type backend4ValueV3 struct {
+	Address   [4]byte
+	Port      uint16
+	Proto     uint8
+	Flags     uint8
+	ClusterID uint16
+	Zone      uint8
+	Pad       uint8
+}
+
+type backend struct {
+	addr util.Address
+	port uint16
+}
+
+type ciliumLoadBalancerConntracker struct {
+	m                  sync.Mutex
+	backends           *maps.GenericMap[backend4KeyV3, backend4ValueV3]
+	ctTCP, ctUDP       *maps.GenericMap[tupleKey4, ctEntry]
+	backendIDToBackend map[uint32]backend
+	stop               chan struct{}
+}
+
+func newCiliumLoadBalancerConntracker(cfg *config.Config) (netlink.Conntracker, error) {
+	if !cfg.EnableCiliumLBConntracker {
+		return netlink.NewNoOpConntracker(), nil
+	}
+
+	ctTCP, err := ebpf.LoadPinnedMap("/sys/fs/bpf/tc/globals/cilium_ct4_global", &ebpf.LoadPinOptions{
```

Comment:
just want to make sure we already had permissions to access this, and if not do we have to make changes in the chart etc

Reply:
sorry I just saw this in the description, is there a ticket for this

Reply:
This was merged today: https://github.com/DataDog/helm-charts/commit/3baa94baafc3145e587825dade7364eb21a21985